en/middleware/xrce_dds.md

```diff
@@ -0,0 +1,303 @@
+:::
+
+To start the agent with settings for connecting to the XRCE-DDS client running on the simulator:
+
+```sh
+MicroXRCEAgent udp4 -p 8888
+```
+
+### Install from Snap Package
+
+Install from a snap package on Ubuntu using the following command:
+
+```sh
+sudo snap install micro-xrce-dds-agent --edge
+```
+
+To start the agent with settings for connecting to the XRCE-DDS client running on the simulator (note that the command name is different than if you build the agent locally):
+
+```sh
+micro-xrce-dds-agent udp4 -p 8888
+```
+
+:::note
+At time of writing the stable of version installed from snap connects to PX4 but reports errors creating topics.
+The development version, fetched using `--edge` above, does work.
+:::
+
+### Build/Run within ROS2 Workspace
+
+The agent can be built and launched within a ROS2 workspace (or build standalone and launched from a workspace.
+You must already have installed ROS 2 following the instructions in: [ROS 2 User Guide > Install ROS 2](##TBDXXX)
+
+To build the agent within ROS:
+
+1. Create a workspace directory for the agent:
+
+   ```sh
+   mkdir -p ~/ws_xrce_dds_agent/src
+   ```
+
+1. Clone the source code for the eProsima [Micro-XRCE-DDS-Agent](https://github.com/eProsima/Micro-XRCE-DDS-Agent) to the `/src` directory (the `main` branch is cloned by default):
+
+   ```sh
+   cd ~/px4_ros_xrce_dds_ws/src
+   git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
+   ```
+1. Source the ROS2 development environment, in this case "ROS 2 foxy", and compile the workspace using `colcon`:
+
+   ```sh
+   cd ..
+   source /opt/ros/foxy/setup.bash
+   colcon build
+   ```
+   This builds all the folders under `/src` using the sourced toolchain.
+
+
+To run the XRCE-DDS agent in the workspace:
+
+1. Source the `local_setup.bash` to make the executables available in the terminal (also `setup.bash` if using a new terminal).
+
+   ```sh
+   source /opt/ros/foxy/setup.bash
+   source install/local_setup.bash
+   ```
+1. Start the agent with settings for connecting to the XRCE-DDS client running on the simulator: 
+
+   ```sh
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+## Starting XRCE-DDS
+
+### Starting the Agent
+
+The agent is used to connect to the client over a particular channel, such as UDP, TCP, or a serial connection.
+The channel settings are specified when the agent is started, using command line options. 
+These are documented in the eProsima user guide: [Micro XRCE-DDS Agent > Agent CLI](https://micro-xrce-dds.docs.eprosima.com/en/latest/agent.html#agent-cli). <!-- what options work? i.e. does PX4 allow CAN? -->
+
+:::note
+You should create a single instance of the agent for each channel over which you need to connect.
+:::
+
+For example, the PX4 simulator runs the XRCE-DDS client over UDP on port 8888, so to connect to the simulator you would start the agent with the command:
+
+```sh
+MicroXRCEAgent udp4 -p 8888
```

Comment:
This is all a bit "thin". Can't really update until we have a working companion setup guide/example.